src/RevisionPopupWidget.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @class
+ * @constructor
+ */
+const RevisionPopupWidget = function RevisionPopupWidget() {
+	// Parent constructor
+	RevisionPopupWidget.parent.call( this, {
+		padded: true,
+		autoClose: true,
+		position: 'above',
+		// FIXME: 'force-left' for RTL languages
+		align: 'force-right',
+		hideWhenOutOfView: false,
+		$content: $( '<div>' )
+			.addClass( 'ext-wwt-revisionPopupWidget-content' )
+	} );
+};
+
+/* Setup */
+OO.inheritClass( RevisionPopupWidget, OO.ui.PopupWidget );
+
+/**
+ * Update the contents of the revision popup.
+ * @param {Object} data As returned by Api.getTokenInfo().
+ */
+RevisionPopupWidget.prototype.setContent = function ( data ) {
+	// Note that the English namespaces will normalize to the wiki's local namespaces.
+	const userLinks = `
```

Comment:
I have to admit, I can't decide what I think about this :D It absolutely 100% might be just personal preference... but would it not be clearer/nicer to jQuerify all of this?

Reply:
Hehe, I almost asked you ahead of time. I started with jQuery then stopped; there's just a lot of markup and attributes. It would make this 4-line template into a really big method (assuming you want `.prop()` etc. to be on its own line). To me it's easier to read the raw markup as a template, similar to Twig. If we were all ES5 then I'd certainly be more inclined to use jQuery :)

I did consider however using an array of templates and joining them, which might make it even easier to read (note that we can't have whitespace between the parentheses and surrounding content):
```
const userLinks = [
    `<a target="_blank" href="${mw.util.getUrl( `User:${data.username}` )}">${data.username}</a>`,
    mw.msg( 'parentheses-start' ),
    `<a target="_blank" href="${mw.util.getUrl( `User talk:${data.username}` )}">${mw.msg( 'talkpagelinktext' )}</a>`,
    ...
].join();
```